config/config.go

```diff
@@ -746,6 +750,39 @@ func (re *Regexp) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return nil
 }
 
+type Matchers []labels.Matchers
+
+func (m *Matchers) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var lines []string
+	if err := unmarshal(&lines); err != nil {
+		return err
+	}
+
+	for _, line := range lines {
+		labelMatchers, err := labels.ParseMatchers(line)
+		if err != nil {
+			return err
+		}
+		for i := range labelMatchers {
+			if labelMatchers[i].Type == labels.MatchRegexp || labelMatchers[i].Type == labels.MatchNotRegexp {
+				re, err := regexp.Compile("^(?:" + labelMatchers[i].Value + ")$")
```

Comment:
No need to compile the regexp. That should all be done by `labels.ParseMatchers(line)` above. I think the only thing left to do here is to append all the `labelMatchers` to `&m`.